src/common.c

```diff
@@ -1137,6 +1137,224 @@ void ofi_byteq_writev(struct ofi_byteq *byteq, const struct iovec *iov,
 	}
 }
 
+#ifdef HAVE_LIBURING
+ssize_t ofi_uring_complete(struct ofi_uring_ctx *uctx)
+{
+	assert(uctx->state == OFI_URING_DONE);
+
+	uctx->state = OFI_URING_IDLE;
+	if (uctx->res < 0) {
+		errno = -uctx->res;
+		return -1;
```

Comment:
Please take a look at the comment I posted [here](https://github.com/ofiwg/libfabric/pull/6761#issuecomment-1285755093) as it describes my intention behind the structure `ofi_uring_ctx`.

`ofi_uring_{send,sendv,sendmsg,recv,recvv,recvmsg}` mimic their non-io_uring counterparts (`ofi_{send,sendv,....}`). 
In the case of a `send` operation, the function returns the number of bytes transferred or -1 if an error occurs (with `errno` is set appropriately). If no data could be transferred (e.g., the corresponding CQE was not yet polled for the given `ofi_uring_ctx` and the status is BUSY) the function would return 0.
The same logic applies for a `recv` operation: the function returns -1 when an error occurs, 0 if the connection has shut down or > 0 to indicate the number of bytes received. If no data could be received (e.g., `ofi_uring_ctx` is BUSY because the CQE was not polled yet), the function would return -1 with `errno=EAGAIN`.

This trick allows us to provide a similar interface between io_uring and non-io_uring implementations. The only exception is that the io_uring implementation requires the separate action of polling the CQ so that `ofi_uring_ctx` can transition from BUSY to DONE, which would allow the send/recv functions to move forward.

Reply:
`uctx->res` contains the status code of the completed operation. It is copied from `io_uring_cqe::res` when a CQE is pulled off the completion queue by the function `ofi_uring_get_completed()` below.